django/db/models/sql/compiler.py

```diff
@@ -2090,6 +2090,62 @@ def pre_sql_setup(self):
         self.query.reset_refcounts(refcounts_before)
 
 
+class SQLBulkUpdateCompiler(SQLCompiler):
+    def execute_sql(self, result_type):
+        cursor = super().execute_sql(result_type)
+        try:
+            rows = cursor.rowcount if cursor else 0
+        finally:
+            if cursor:
+                cursor.close()
+        return rows
+
+    def get_update_clause(self, values):
+        qn = self.quote_name_unless_alias
+        return "UPDATE %s " % qn(self.query.base_table), tuple()
+
+    def get_subquery_from_clause(self, values):
+        values_sql, value_params = self.compile(values)
+        return "FROM (%s) AS subquery" % values_sql, tuple(value_params)
+
+    def get_set_sql_clause(self, values):
+        qn = self.quote_name_unless_alias
+        set_sql = []
+        for idx, field in enumerate(values.field_list, 2):
```

Comment:
I think a keyword argument is clearer:

```suggestion
        for idx, field in enumerate(values.field_list, start=2):
```

Also, why do we start at `2`here?

Reply:
It’s confusing ! I’m going to add comments and docs around this.

Essentially, this represents a `VALUES` expression with unnamed columns (aka how Postgres and SQLite do things). These are sets of 1-indexed columns. 

The first column is the primary key, and the rest represent the actual values we are updating. So we’re just skipping the primary key column and not updating it.

I’ll probably refactor this to be simpler, and maybe separate the primary key field from the rest.

On MySQL, the column names of a `VALUES` expression are the *first* rows actual, literal values. I’m genuinely impressed at how unbelievably stupid this is. However the workaround is somewhat simple: a named SELECT with a UNION for the rest of the values.